reviewfile.cs

```diff
@@ -0,0 +1,387 @@
+                    } : null
+            });
+        }
+
+        [HttpPut("{opNumber}/eligibility")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        //[Authorize(SGPrepPolicies.CorrectOperationRolePolicy)]
+        public async Task<IActionResult> SaveEligibility([FromBody] EligibilitySubmissionRequest request, string opNumber)
+        {
+            ValidateUserContext();
+            
+            var streamId = opNumber.ToOperationStreamIdentifier();
+            var existingOperation = await _repo.GetOptionalAsync(streamId);
+
+            if (!existingOperation.HasValue)
+            {
+                return NotFound();
+            }
+
+            if (existingOperation.Value.EligibilitySubmission is not null)
+            {
+                throw new EligibilitySubmissionModelInvalidException("An eligibility submission already exists for this operation");
+            }
+
+            string resultValidation = string.empty;
+
+            switch (request.DocType)
+            {
+                case "PP":
+                    resultValidation = await ValidatePPSubmissionRequestAsync(request, opNumber);
+                    switch (request.DocVersion)
+                    {                        
+                       case "Approval":
+                        await SendAproval(opNumber);
+                        break;
+                       case "Revision":
+                        await SendRevision(opNumber);
+                    }
+                break;
+                case "Minutes":
+                    resultValidation = await ValidateMinutesSubmissionRequestAsync(request, opNumber);
+                break;
+                case "Annex":
+                    resultValidation = await ValidateAnnexSubmissionRequestAsync(request, opNumber);
+                break;                
+            }  
+
+            if (!string.IsNullOrEmpty(resultValidation))
+            {
+                return BadRequest(resultValidation);
+            }
+
+            PrepareEligibilitySubmissionForSaving(request, existingOperation.Value);
+
+            await _repo.SaveAllUnitsOfWorkAsync();
+
+            return Created(nameof(GetEligibility), existingOperation.Value.EligibilitySubmission);
+        }
+
+        [HttpPost("{opNumber}/eligibility")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        //[Authorize(SGPrepPolicies.CorrectOperationRolePolicy)]
+        public async Task<IActionResult> UpdateEligibility([FromBody] EligibilitySubmissionRequest request, string opNumber)
+        {
+            // TODO: Future -- create UserContextValidator and then can make the validator below a composite validator that includes that validator
```

Comment:
This should be added to the backlog instead of a random TODO comment which many people might not understand, If it's a simple change we might consider doing it now instead